extensions/narayana-jta/runtime/src/main/java/io/quarkus/narayana/jta/runtime/NarayanaJtaRecorder.java

```diff
@@ -32,34 +47,53 @@ public void setNodeName(final TransactionManagerConfiguration transactions) {
         }
     }
 
-    public void setDefaultProperties(Properties properties) {
-        //TODO: this is a huge hack to avoid loading XML parsers
-        //this needs a proper SPI
-        try {
-            Field field = PropertiesFactory.class.getDeclaredField("delegatePropertiesFactory");
-            field.setAccessible(true);
-            field.set(null, new QuarkusPropertiesFactory(properties));
-
-        } catch (Exception e) {
-            log.error("Could not override transaction properties factory", e);
-        }
-
-        defaultProperties = properties;
-    }
-
     public void setDefaultTimeout(TransactionManagerConfiguration transactions) {
         transactions.defaultTransactionTimeout.ifPresent(defaultTimeout -> {
             arjPropertyManager.getCoordinatorEnvironmentBean().setDefaultTimeout((int) defaultTimeout.getSeconds());
             TxControl.setDefaultTimeout((int) defaultTimeout.getSeconds());
```

Comment:
with this change the `TxControl.setter` can't be called here. When it's called then `TxControl` stays frozen at current state. No changes into beans properties are not taken into consideration.

Reply:
This is in the recorder phase during build time so it gets baked into the executable.

Reply:
-1

Reply:
@mmusgrov is that for sure? Have you considered that the `@BuildStep` you refer to is annotated with `@Record(RUNTIME_INIT)`?

Reply:
I am not clear on your issue with this code: when the app starts up the default timeout value is read from the quarkus confg and that value is then set on the bean and on the TxControl. The bytecode for doing this is run from the main method and therefore the default timeout will have been set when the application starts running transactions.

The timeout value in TxControl is not marked as `final`. When TxControl.setDefaultTimeout() is called the class will be initialized and it is initialized with the correct value for default timeout as read from the quarkus config.

Reply:
@ochaloup ^^ this comment still needs your input.

Reply:
ah, right. I missed your response, sorry. 

You are right. My first response was on PR as it was in November and my second one was a reaction to statement that all is baked into executable.

As the code is now you are right the the `TxControl.setDefaultTimeout()` is called as the last call in the quarkus `setConfig` method. The only trouble I can see is if the methods are reordered in manner that the `disableTransactionStatusManager()` is called after `setDefaultTimeout(transactions)` in the quarkus `setConfig` method.

From what I can see my comment regarding the `TxControl` is not relevant in the current state.